Ony 72 add aws config

abdullahcevo opened the pull request:
This PR adds Functionality to receive alerts from Eventbridge to Slack. These types of alerts are configured:
1. When AWS Config Compliance for resources Changes. Todo is filter notification state of resources changes to Non-Compliant. 
2.  When auto-remediation actions fails.

Also added the parsers to parse respective messages, and convert them to messages format accepted by Slack API.

Changed Nodejs version of Lambda from 10.x to 22.x runtime. Without this change, the Cloudformation Stack for Lambda will fail to deploy.

abdullahshafiq1 committed:
added parser for event-bridge aws config notifications

abdullahshafiq1 committed:
added PR template

abdullahshafiq1 committed:
updated parsers list

abdullahshafiq1 committed:
update nodejs to 22.x

abdullahshafiq1 committed:
update order in parser list

abdullahshafiq1 committed:
updated parser matcher and fields

abdullahshafiq1 committed:
updated parser matcher

abdullahshafiq1 committed:
fix aws-config parser

abdullahshafiq1 committed:
fix aws-config parser color formatting

abdullahshafiq1 committed:
added new fields - modified visual layout

abdullahshafiq1 committed:
added config reevaluation parser

abdullahshafiq1 committed:
updated parsers list

abdullahshafiq1 committed:
updating documentation

abdullahshafiq1 committed:
updating documentation

abdullahcevo committed:
Update README.md

updated documnetation

abdullahshafiq1 committed:
eventbridge aws config stack

abdullahshafiq1 committed:
updating documentation

abdullahshafiq1 committed:
updating documentation

abdullahshafiq1 committed:
updating documentation

abdullahshafiq1 committed:
temp cfn

abdullahshafiq1 committed:
remove role name

abdullahshafiq1 committed:
added parameters

matthew-oneill reviewed (changes requested):
Looking good @abdullahcevo,

Could I recommend that it might be worth providing some docs, possible in the eventbridge folder, about the eventbridge CFN templates. Something that includes details of why these are here, what they should be used for, how to deploy, etc. 

They don't currently follow the pattern of the existing code, and don't appear to be deployed as part of the standard deployment process so we will need some details about these for anyone who looks at the codebase in the future.

abdullahshafiq1 committed:
change rule name to DeployAllWSConfigComplianceChangesRule

abdullahshafiq1 committed:
change rule name to DeployAllWSConfigComplianceChangesRule

abdullahshafiq1 committed:
added rule DetectAWSConfigNonComplianceChanges

abdullahshafiq1 committed:
adding doco for eventbridge folder

abdullahshafiq1 committed:
update doco for eventbridge folder

abdullahshafiq1 committed:
update params

abdullahshafiq1 committed:
update eventbridge readme

abdullahcevo commented:
Thanks @matthew-oneill .. I have added another README.md in eventbridge folder with instructions. I've also split the event-bridge rules into two i. DetectAllAWSConfigComplianceChanges (For detecting All Complaince Changes), ii. DetectAWSConfigNonComplianceChanges (For detecting only non compliant resources). I''ve added parameters in the CFN to switch on/off these rules.

matthew-oneill commented on eventbridge/README.md:
```diff
@@ -0,0 +1,39 @@
+[![license](https://img.shields.io/github/license/arabold/aws-to-slack.svg)](https://github.com/arabold/aws-to-slack/blob/master/LICENSE)
+
+
+This document is specific to Cloudformation Stacks that reside under *eventbridge* directory. This stack deploys the following components with certain parameters that are configurable.
```
Might be worth also mentioning the explicit purpose of these templates (i.e. they are to assist with getting AWS Config notifications into Slack).

Also detail why there are two different templates, what their uses are, etc.

matthew-oneill commented on eventbridge/temp-eventbridge.yaml:
```diff
@@ -0,0 +1,136 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Description: CloudFormation template for EventBridge Events and associated Role to send to SNS When AWS Config Compliance Changes or Remediation Actions Fail
```
The description for this template is the same as the event-bridge-cfn.yaml

matthew-oneill commented on eventbridge/README.md:
```diff
@@ -0,0 +1,39 @@
+
+
+## How to Deploy: 
+1. Go to your Management/Audit/Security AWS account from where you wish to deploy the SNS Topic, Eventbridge Rules and required IAM role.
```
As discussed in stand up can we please provide more detail here.

matthew-oneill replied:
@abdullahcevo,

Let's provide details here about how to specifically deploy the eventbridge rules for AWS Config monitoring. Could just copy and paste from the README, or even better direct the user to the README at the top level for deployment instructions so this only needs to be maintained in one place.

abdullahcevo replied:
Done

abdullahshafiq1 committed:
update eventbridge doc

abdullahcevo committed:
Delete eventbridge/temp-eventbridge.yaml

delete temporary cfn file

abdullahcevo committed:
Update instructions to create Slack App and Webhooks

abdullahcevo committed:
Update how to Create Slack app and webhooks

matthew-oneill commented on eventbridge/README.md:
```diff
@@ -0,0 +1,41 @@
+This stack deploys the following components with certain parameters that are configurable.
+
+## Template Parameters
+  1. DeployAllWSConfigComplianceChangesRule (Boolean). Controls creation of Eventbridge rule to detect all Compliance Changes. Default = true.
```
When we say compliance changes can we be specific that these are AWS Config changes.

abdullahcevo replied:
Done.

abdullahshafiq1 committed:
upating documentation

matthew-oneill commented on DOCUMENTATION.md:
```diff
@@ -0,0 +1,157 @@
+# AWS-to-Slack Integration
+
```
Why does this file exist? Shouldn't this all be covered in the README.md?

abdullahshafiq1 committed:
update document

abdullahshafiq1 committed:
upate README

abdullahshafiq1 committed:
upadte README

abdullahshafiq1 committed:
upadte README

abdullahshafiq1 committed:
upadte README

abdullahshafiq1 committed:
upadte README

abdullahshafiq1 committed:
upadte README

abdullahshafiq1 committed:
upadte README

abdullahshafiq1 committed:
upadte README

abdullahshafiq1 committed:
upadte README

abdullahcevo committed:
Delete DOCUMENTATION.md

delete extra file

matthew-oneill commented on README.md:
```diff
@@ -45,6 +42,147 @@ Supported AWS product notification formats:
+2. Grab the Cloudformation template under eventbridge directory, and deploy it as StackSet. This template requires parameters below:
+
+- **DeployAllWSConfigComplianceChangesRule (Boolean)**. Controls creation of Eventbridge rule to detect all Compliance Changes. Default = true.
+- **DeployAWSConfigNonComplianceAlertRule:** Controls creation of Eventbridge rule to only when resources become non-Compliant. Default = true. 
```
Are the above two rules in conflict?

Say for example if I set the "DeployAllAWSConfigComplianceChangeRule" to true, but the "DeployAWSConfigNonComplianceAlertRule" to false, would I expect to get alerts for non-compliant config events?

abdullahcevo replied:
Clarification Added .

matthew-oneill commented on README.md:
```diff
@@ -45,6 +42,147 @@ Supported AWS product notification formats:
+
+3. Note the SNS topic ARN from the deployment output for use in the central account configuration
+
+### Step 3: Deploy main Cloudformation Stack in Central Account (Audit/Management)
```
Add some detail about what the Central Account is. 

This should be the account where you intend the messages to be handled and sent to slack from. So it doesn't need to be the management account nor the audit account, but whichever account you choose.

matthew-oneill commented on README.md:
```diff
@@ -45,6 +42,147 @@ Supported AWS product notification formats:
+
+
+
+## WARNING ------------------  Below steps are deprecated and need to be deleted.
```
Is there a reason we are keeping them if they need to be deleted? Why not just delete now?

abdullahcevo replied:
Deleted.

matthew-oneill commented on README.md:
````diff
@@ -45,6 +42,147 @@ Supported AWS product notification formats:
+     --stack-name aws-to-slack-events
+   ```
+
+#### Deployment as StackSet in Management / Audit Account. ( This should be delegated Admin for Org.)
````
This appears to be the steps required for AWS Config monitoring deployment ONLY. Can we please make clear which steps are ONLY required if you intend to set up AWS Config monitoring.

abdullahcevo replied:
Done

abdullahshafiq1 committed:
redirect to main README

abdullahshafiq1 committed:
clarifications

abdullahshafiq1 committed:
clarifications

abdullahshafiq1 committed:
removed old doc

abdullahshafiq1 committed:
Merge branch 'ONY-72-Add-AWS-Config' of https://github.com/cevoaustralia/aws-to-slack into ONY-72-Add-AWS-Config

matthew-oneill commented:
Closing Pull Request.

Going to separate this out as its own repository to act as a plug-in for the aws-to-slack stack, not as part of it.
